substrate/codec/src/slicable.rs

```diff
@@ -92,18 +110,60 @@ impl Slicable for Vec<u8> {
 	}
 }
 
+macro_rules! impl_vec_simple_array {
+	($($size:expr)*) => {
+		$(
+			impl<T> Slicable for Vec<[T; $size]>
+				where [T; $size]: EndianSensitive
+			{
+				fn decode<I: Input>(input: &mut I) -> Option<Self> {
+					u32::decode(input).and_then(move |len| {
+						let mut r = Vec::with_capacity(len as usize);
+						for _ in 0..len {
+							r.push(match Slicable::decode(input) {
+								Some(x) => x,
+								None => return None,
+							});
+						}
+
+						Some(r)
+					})
+				}
+
+				fn as_slice_then<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+					f(&self.to_vec())
+				}
+
+				fn to_vec(&self) -> Vec<u8> {
+					use rstd::iter::Extend;
+
+					let len = self.len();
+					assert!(len <= u32::max_value() as usize, "Attempted to serialize vec with too many elements.");
+
+					let mut r: Vec<u8> = Vec::new().join(&(len as u32));
+					for item in self {
+						r.extend(item.to_vec());
+					}
+					r
+				}
+			}
+		)*
+	}
+}
+
+impl_vec_simple_array!(1 2 4 8 16 32 64);
```

Comment:
i think convention is to delimit items in lists with `,`s rather than just whitespace

Reply:
there is lots of precedent, e.g. https://doc.rust-lang.org/stable/src/core/cmp.rs.html#782

Reply:
the other way too, even as part of this PR in this very same file:

> `tuple_impl!(A, B, C, D, E, F, G, H, I, J, K,);`